lib/ansible/module_utils/network/nxos/nxos.py

```diff
@@ -726,33 +726,46 @@ def feature_enable(self):
                 ref['_cli_is_feature_disabled'] = ref['_proposed']
 
 
-    def get_platform_id(self):
-        """Query device for platform type"""
-        info = get_capabilities(self._module).get('device_info')
-        os_platform = info.get('network_os_platform')
-        plat = None
-
-        # Supported Platform IDs (N35 N3K N5K N6K N7K N9K)
-        if not os_platform:
+    def get_platform_shortname(self):
+        """Query device for platform type, normalize to a shortname/nickname.
+        Returns platform shortname (e.g. 'N3K-3058P' returns 'N3K') or None.
+        """
```

Comment:
Do y'all have any UT automation (or doctests) for this module? This function could really use a test matrix of sample platform outputs and the expected return value.

Reply:
Not yet. I still have to write the UT's. I'll do that next and open a separate PR.